### PR TITLE
New command "show-coverage"

### PR DESCRIPTION
**Pull request by caiosba**
We wanted to show the coverage results easily in our Travis job output log, and we had two parallel jobs, one with the unit tests and the other one with the functional tests. We use the `sum-coverage` command to combine those two into a single coverage file.

This PR adds a new command `show-coverage` which parses that single coverage file and outputs the coverage information, for example, coverage percentage, number of covered lines, total number of lines, number of missing lines and which lines from which files are not covered. Example output:

```
Coverage: 99.55% (10531/10579 lines covered, 48 missing)
Uncovered lines by file:
lib/check_config.rb: 4, 5
app/controllers/concerns/twitter_authentication.rb: 6, 8, 10
```

### CODE REVIEW

**Comment by efueger** [comment]
Hi @caiosba - thanks for the PR. I've just put this in the queue for our engineers to review. 👌

File: cmd/show-coverage.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    dat, err := ioutil.ReadFile(args[0])
+    if err != nil {
+      return errors.New("could not open input file")
+    }
+
+    var result map[string]interface{}
+    json.Unmarshal([]byte(string(dat)), &result)
+    line_counts := result["line_counts"].(map[string]interface{})
+    header := "Coverage: %.2f%% (%d/%d lines covered, %d missing)"
+    fmt.Println(fmt.Sprintf(header, result["covered_percent"], int(line_counts["covered"].(float64)), int(line_counts["total"].(float64)), int(line_counts["missed"].(float64))))
```
Comment (filipesperandio) [inline cmd/show-coverage.go]: The body of this function is a bit dense. Do you mind extracting a few new functions to smooth out the read?

Comment (caiosba) [reply cmd/show-coverage.go]: Done

File: cmd/show-coverage.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      fmt.Println("Uncovered lines by file:")
+      files := result["source_files"].([]interface{})
+
+      for _, file_obj := range files {
+        file := file_obj.(map[string]interface{})
+        if file["covered_percent"].(float64) < 100 {
+          printUncoveredLinesFromFile(file)
+        }
+      }
+    }
```
Comment (filipesperandio) [inline cmd/show-coverage.go]: What about another one here? 😁

Comment (caiosba) [reply cmd/show-coverage.go]: Another one what? :grin:

Comment (filipesperandio) [reply cmd/show-coverage.go]: function!
I would say something like `printUncoveredLines`?

Comment (caiosba) [reply cmd/show-coverage.go]: Ah yes, makes sense... done! :)

**Comment by efueger** [comment]
Looks like both sets of tests failed:

image

**Comment by caiosba** [comment]
> Looks like both sets of tests failed:
> 
> image

I can't see which part of my code would trigger this failure... can I be sure it's working for the main branch? BTW the failure I see is that the double quotes are escaped.

**Comment by caiosba** [comment]
I'll try to run the tests locally to see.

**Comment by caiosba** [comment]
I was able to reproduce the test failure by running them in the `circleci/golang:latest` Docker image, and the same test fails on `master` as well. But I just fixed it in this PR's branch.

**Comment by filipesperandio** [comment]
@caiosba thanks for looking into it, wasn't finding the time to dig.
Yeah, looks like the underlying dependencies got in the way and the fix seems fine. 👍

**Pull request merged**